### PR TITLE
Scale RSS feed query to any episode count (#865)

### PR DESCRIPTION
**Pull request by zahardev**
## Summary
- Podcasts with a very large number of episodes (thousands) had a broken RSS feed — it returned **no episodes** because the database query grew too large and the server aborted it. This fixes that: the feed now works no matter how many episodes a podcast has.
- The episodes shown in the feed are **exactly the same as before** — same episodes, same order, same "max episodes" limit. This only changes *how* the feed is fetched, not *what* it returns.
- Bonus fix for an edge case: on sites that turn other post types into episodes, posts without audio could slip into the feed when nothing had audio. They no longer do.

## Test plan
1. **The feed shows the same episode IDs as before (most important).** Open a podcast's RSS feed and note the episodes it lists (GUIDs/titles, their order, and the count). Compare against what the feed returned before this change — it must be identical: same episodes, same order, same number (capped at the "Number of episodes in feed" setting).
2. **A large podcast's feed now loads.** On a podcast with a very large episode count (or one that previously returned an empty/broken feed), open the feed and confirm episodes now appear — no empty or errored feed.
3. **(Only if "use posts as episodes" is enabled)** Confirm only posts that have an audio file appear in the feed; with that setting off, all published episodes appear — same as before.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved episode feed generation for better scalability with large episode libraries.
  * Ensured feeds return published episodes while excluding drafts and private content.
  * Improved filtering for additional podcast content, including audio availability and series selection.
  * Preserved support for feed limits, ordering, and empty-feed scenarios.
  * Maintained compatibility for episode ID lists used in dashboard summaries.

* **Tests**
  * Added comprehensive automated coverage for episode queries and feed behavior.

### CODE REVIEW

File: tests/WPUnit/SspEpisodesQueryTest.php [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+			'post_date_gmt' => '2025-06-01 09:00:00',
+		) );
+
+		$ids = $this->feed_ids();
+
+		$this->assertLessThan(
+			array_search( $older, $ids, true ),
+			array_search( $newer, $ids, true ),
+			'Newer episode should sort before older by post date'
+		);
```
[the remark this answers is not in the document]
Comment (zahardev) [reply tests/WPUnit/SspEpisodesQueryTest.php:269]: Good catch — fixed in 1429a6ab. Added `assertContains()` for both compared episodes before the `array_search()`/`assertLessThan()` position check in all three ordering tests, so a dropped episode now fails instead of silently passing.
